### DISCUSSION TITLE
Two questions on auth

### DISCUSSION THREAD

**Discussion by heauton** [supabase/supabase#1572] [Questions] [answered]
- Can we generate a magic link URL without actually sending emails?
- Can we set up a custom URL endpoint? (e.g. xxx.supabase.co -> api.myapp.com)

**Comment by quicksnap** [comment, marked as answer]
I think being able to generate a magic link without involving the user would be good for development, but it's essentially a JWT auth token, allowing the link-holder to auth as the user. Might not be a great idea security-wise. I'm also not aware of such functionality within hosted Supabase.

From my exploration of hosted Supabase, there's no CNAME tools to avoid the xxx.supabase.co magic link. It'd be a nice feature, for sure.

**Reply by heauton** [reply]
Yeah, generating a magic link should happen on the server-side, which would defeat the simplicity Supabase brings. It may be helpful for more advanced use cases. Thanks for sharing your experience!

**Reply by awalias** [reply]
for magic links I've opened an issue here: supabase/gotrue#104

the custom domains in emails should be solved by this one: supabase/gotrue#88

looking for help on these two if anyone is familiar with go
